### PR TITLE
deploy(vars): support of the alternative layout approach

### PR DESCRIPTION
**Pull request by maria-Petrova**
The PR aims to support the alternative layout approach for the upcoming releases.

### CODE REVIEW

**Review by napetrov — APPROVED** [review]

File: cmake/templates/oneDALConfig.cmake.in [hunk 1]
```diff
@@ -91,7 +91,12 @@ if (@VERSIONS_SET@ STREQUAL "TRUE")
     set(_dal_ver_major_bin @DAL_VER_MAJOR_BIN@)
     set(_dal_ver_minor_bin @DAL_VER_MINOR_BIN@)
 elseif (@VERSIONS_SET@ STREQUAL "FALSE")
-    file(READ ${_dal_root}/include/services/library_version_info.h DAL_VERSION_INFO)
+    # version_info.h is set according to the relevant package structure 
+    set(version_info.h ${_dal_root}/include/services/library_version_info.h)
+    if (NOT EXISTS "${version_info.h}")
+        set(version_info.h ${_dal_root}/include/dal/services/library_version_info.h)
+    endif()
```
Comment (KulikovNikita) [inline cmake/templates/oneDALConfig.cmake.in:98]: What about just using `if() ... else() ... endif()` statements?

File: deploy/local/vars_lnx.sh [hunk 2]
```diff
@@ -213,21 +213,31 @@ fi
...
+    export DAL_MINOR_BINARY=__DAL_MINOR_BINARY__
+    export DALROOT=$__daal_tmp_dir
+    export DAALROOT=$__daal_tmp_dir
+    export CPATH=$__daal_tmp_dir/include${CPATH+:${CPATH}}
+    export LIBRARY_PATH=$__daal_tmp_dir/lib/intel64${LIBRARY_PATH+:${LIBRARY_PATH}}
+    export LD_LIBRARY_PATH=$__daal_tmp_dir/lib/intel64${LD_LIBRARY_PATH+:${LD_LIBRARY_PATH}}
+    export CLASSPATH=$__daal_tmp_dir/lib/onedal.jar${CLASSPATH+:${CLASSPATH}}
+    export CMAKE_PREFIX_PATH=$__daal_tmp_dir${CMAKE_PREFIX_PATH+:${CMAKE_PREFIX_PATH}}
+    export PKG_CONFIG_PATH=$__daal_tmp_dir/lib/pkgconfig${PKG_CONFIG_PATH+:${PKG_CONFIG_PATH}}
+  ;;
```
Comment (KulikovNikita) [inline deploy/local/vars_lnx.sh:237]: What do these symbols do?

**Pull request merged**